frontend/internal-packages/agent/src/workflow/nodes/validateInitialSchemaNode.integration.test.ts

```diff
@@ -0,0 +1,270 @@
+import { AIMessage, HumanMessage } from '@langchain/core/messages'
+import { END, START, StateGraph } from '@langchain/langgraph'
+import { aColumn, aSchema, aTable } from '@liam-hq/schema'
+import { describe, expect, it } from 'vitest'
+import { getTestConfig } from '../../../test-utils/workflowTestHelpers'
+import type { WorkflowState } from '../../types'
+import { WorkflowTerminationError } from '../../utils/errorHandling'
+import { workflowAnnotation } from '../../workflowAnnotation'
+
+import { validateInitialSchemaNode } from './validateInitialSchemaNode'
+
+describe('validateInitialSchemaNode Integration', () => {
+  describe('First execution scenarios', () => {
+    it('should display fresh database message when schema is empty', async () => {
+      // Arrange
+      const graph = new StateGraph(workflowAnnotation)
+        .addNode('validateInitialSchema', validateInitialSchemaNode)
+        .addEdge(START, 'validateInitialSchema')
+        .addEdge('validateInitialSchema', END)
+        .compile()
+
+      const { config, context } = await getTestConfig()
+
+      const state: WorkflowState = {
+        messages: [new HumanMessage('Create a new database schema')],
+        designSessionId: context.designSessionId,
+        organizationId: context.organizationId,
+        userId: context.userId,
+        userInput: 'Create a new database schema',
+        schemaData: aSchema({
+          tables: {},
+          enums: {},
+          extensions: {},
+        }),
+        analyzedRequirements: undefined,
+        testcases: [],
+        buildingSchemaId: 'test-building-schema-id',
+        latestVersionNumber: 1,
+        next: 'leadAgent',
+      }
+
+      // Act
+      const stream = await graph.stream(state, config)
+      const results = []
+      for await (const result of stream) {
+        results.push(result)
+      }
```

Comment:
In this case, since we don't need to output to a stream for debugging, I think it's clearer to receive the state via invoke. The following code is conceptual and has not been tested.

```suggestion
      const result = await graph.invoke(state, config)
      expect(result).toEqual(state) // Nothing has changed at present. Messages will be added in the future.
```

Reply:
You were absolutely right!
 I've made the corrections below.
https://github.com/liam-hq/liam/pull/3359/commits/6c7fe800a1b382e328dae2fa8c8d2396d2adf046